Implement LRU Cache with put and get functionality

kirti734 opened the pull request:
This implementation provides a Least Recently Used (LRU) cache with basic operations such as put and get. It includes a linked list for maintaining the order of usage and a hashmap for quick access.

kirti734 committed:
Implement LRU Cache with put and get functionality

This implementation provides a Least Recently Used (LRU) cache with basic operations such as put and get. It includes a linked list for maintaining the order of usage and a hashmap for quick access.

priyankagupta-ITT commented on LRU_CACHE_Implementation.c:
1. for put

2. for get

3. for exit

Enter choice: 1

Enter the key: 101
Enter the value: d ff

1. for put

2. for get

3. for exit

Enter choice:
Choice should be numeric value

1. for put

2. for get

3. for exit

Handle this scenario

kirti734 replied:
I handle the test case using in my function by:
if(fgets(value, VALUE_LENGTH, stdin) == NULL)
	{
		printf("\nFailed to read value\n");
		return;
	}
	value[strcspn(value, "\n")] = '\0';

priyankagupta-ITT commented on LRU_CACHE_Implementation.c:
```diff
@@ -0,0 +1,368 @@
+	}
+}
+
+void insertPut()
```
refactor this method.

kirti734 replied:
I have refactored the insertPut() method with better input validation and value handling.
void insertPut()
{
	int key = 0;
	printf("\nEnter the key: ");
	if (scanf("%d", &key) != 1)
	{
		printf("\nKey should be numeric\n");
		while(getchar() != '\n');
		return;
	}
	while(getchar() != '\n');
	
	char value[VALUE_LENGTH];
	printf("Enter the value: ");
	if(fgets(value, VALUE_LENGTH, stdin) == NULL)
	{
		printf("\nFailed to read value\n");
		return;
	}
	value[strcspn(value, "\n")] = '\0';
	
	if(strlen(value) == 0)
	{
		printf("\nPlease enter the value\n");
		return;
	}
	
	put(key, value);
}
is this correct? Please tell me how can I improve this.

priyankagupta-ITT replied:
refactor the method name.

kirti734 replied:
Yes, I refracted the method name by getUserInputForPut() from insertPut()
also I updated the getUserInputForGet() from insertGet()

kirti734 committed:
Improve value input handling in LRU cache

Updated input handling for value entry to use fgets, ensuring proper error handling and trimming of newline characters.

kirti734 committed:
Rename insertPut and insertGet functions
